app/controllers/application_controller.rb

```diff
@@ -2,9 +2,126 @@
 
 class ApplicationController < Sinatra::Base
 
-  configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
-  end
+	configure do
+		set :public_folder, 'public'
+		set :views, 'app/views'
+		enable :sessions
+	end
 
-end
\ No newline at end of file
+	get '/' do
+		erb :index
+	end
+
+	get '/signup' do
+		if !session[:id].nil?
+			redirect '/tweets'
+		end
+		erb :'/users/create_user'
+	end
+
+	post '/signup' do
+		@user = User.create(params)
+		if params[:username] == "" || params[:password] == "" || params[:email] == ""
+			redirect '/signup'
+		end
+
+		if @user.save
+			session[:id] = @user.id
+			redirect '/tweets'
+		else
+			redirect "/signup"
+		end
+	end
+
+	post "/login" do
+		@user = User.find_by(username: params[:username])
+		if @user && @user.authenticate(params[:password])
+			session[:id] = @user.id
+			redirect "/tweets"
+		else
+			redirect "/failure"
+		end
+	end
+
+	get '/tweets' do
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).